app/controllers/users_controller.rb

```diff
@@ -1,3 +1,30 @@
 class UsersController < ApplicationController
-  def user; end
+  def show
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      render :error
+    else
+      render :show
+    end
```

Comment:
thay đổi code if - else từ 4 - 8 thành 
`return if @user`
`flash[:warning] = "User not found" # dung I18n nhe`
`redirect_to root_path`